### ISSUE TITLE
[BUG]: Ant + Jacoco: 'Failed to enable code coverage: TypeError: Cannot read properties of undefined (reading 'split')' 

### ISSUE THREAD

**Issue by ataverascrespo** [microsoft/azure-pipelines-tasks#19724]
### New issue checklist

- [X] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [X] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [X] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

Ant@V1

### Task version

1.231.1

### Issue Description

When creating an Azure DevOps build pipeline, and working with Ant@V1 task, enabling JaCoCo code coverage is failing, and is generating the following error:

**##[warning]Failed to enable code coverage: TypeError: Cannot read properties of undefined (reading 'split')**

This error was generated with the following YAML configuration for the Ant@V1 task:

```shell
codeCoverageToolOptions: 'JaCoCo'
codeCoverageClassFilesDirectories: 'build/src/'
``` 
After several days of debugging, our team found the issue was able to be resolved by adding the following optional field to the YAML configuration for the Ant@V1 task:
```shell
codeCoverageSourceDirectories: 'components/src/'
```
This is inconsistent with the Ant@V1 documentation, which states the `codeCoverageSourceDirectories` input is optional when `codeCoverageToolOptions != None`. 

Even though it is working with the inclusion of the `codeCoverageSourceDirectories`, this should be looked into as to not cause confusion or issues with Ant tasks on pipelines

### Environment type (Please select at least one enviroment where you face this issue)

- [X] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Redhat Enterprise Linux 7

### Relevant log output

```shell
masking file paths with (...)

##[debug]classFilter=undefined
##[debug]classFilesDirectories=build/src/
##[debug]srcDirectories=undefined
##[debug]rm -rf (...)/cobertura.ser
##[debug]rm -rf (...)/CCReport43F6D5EF
##[debug]removing directory
##[debug]rm -rf (...)/CCReportBuildA4D283EG.xml
##[debug]removing file
##[debug]rm -rf (...)/InstrumentedClasses
##[debug]Input parameters: {"buildfile":"(...)/build.xml","classfilesdirectories":"build/src/","summaryfile":"summary.xml","reportdirectory":"(...)/CCReport43F6D5EF","ccreporttask":"CodeCoverage_9064e1d0","reportbuildfile":"(...)/CCReportBuildA4D283EG.xml"}
##[debug]Extracting Azure Pipelines filter: undefined
##[debug]Applying the filter pattern:  op: 
##[debug]Applying the filter pattern:  op: 
##[debug]Reading XML file: (...)/build.xml
##[debug]Converting XML file to JSON
##[warning]Failed to enable code coverage: TypeError: Cannot read properties of undefined (reading 'split')
##[debug]Processed: ##vso[task.issue type=warning;]Failed to enable code coverage: TypeError: Cannot read properties of undefined (reading 'split')
```

### Full task logs with system.debug enabled

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Repro steps

```yml
- stage: Test
  jobs:
  - job: JUnits
    steps:
    - task: Ant@1
      inputs:
          buildFile: '$(Build.SourcesDirectory)/build.xml'
          options: 
          targets: 'test'
          publishJUnitResults: true
          testResultsFiles: '$(Build.SourcesDirectory)/reports/**/TEST-*.xml'
          codeCoverageToolOptions: 'JaCoCo'
          codeCoverageClassFilesDirectories: 'build/src/'
          javaHomeOption: 'JDKVersion'
          jdkVersionOption: '1.8'
```

**Comment by ataverascrespo** [comment]
Adding this as a comment, as to not make the description too long.

I looked into the JaCoCo code coverage functions and believe I found the root cause of the issue in the `enableCodeCoverage(`) function in the [jacoco.ant.cc.enabler](https://github.com/microsoft/azure-pipelines-tasks-coverage-tools/blob/6f5d1404a02d85af4e11733b43399aaa0d16cab3/jacoco/jacoco.ant.ccenabler.ts) class. 

In `enableCodeCoverage()`, the global `sourceDirs` variable is set to the value of `ccProps["sourcedirectories"]`, which contains the value of the `INPUT_SRCDIRECTORIES` environment variable passed from the Ant task. 

From there, `addCodeCoverageData()` is called, which then calls `getSourceFilter()`. 
```TypeScript
    protected getSourceFilter(): string {
        let srcData = "";
        let srcDirs = this.sourceDirs === null ? "" : this.sourceDirs;
        srcDirs.split(",").forEach(dir => {
            if (!util.isNullOrWhitespace(dir)) {
                srcData += `<fileset dir="${dir}"/>`;
                srcData += os.EOL;
            }
        });
        if (util.isNullOrWhitespace(srcData)) {
            srcData = `<fileset dir="."/>`;
            srcData += os.EOL;
        }
        return srcData;
    }
```
If the `codeCoverageSourceDirectories` YAML input is not given a value, then `INPUT_SRCDIRECTORIES` is undefined, meaning the value of global var `sourceDirs` is set to undefined. Thus, calling split() on an undefined value generates the error and causes JaCoCo code coverage to fail on the pipeline.

This is a simple fix. The issue is the azure-pipelines-tasks-coverage-tools [repo](https://github.com/microsoft/azure-pipelines-tasks-coverage-tools/tree/6f5d1404a02d85af4e11733b43399aaa0d16cab3) (which contains those JaCoCo code coverage functions) is archived, so I can't open any new issues or create any PRs. I'm opening the bug here, hoping that a contributor with access to the private repo will see it.

A minimum recreation of a possible fix would look something like:
```TypeScript    
    let srcData = "";
    let srcDirs = sourceDirs === null ? "" : sourceDirs;
    if (srcDirs) {
        srcDirs.split(",").forEach(dir => {
               // continues if srcDirs != undefined
        });
    }
    else {
        // continues if srcDirs == undefined
    }
    
``` 
After a brief search, this error might also occur for JaCoCo code coverage on Maven and Gradle tasks.

@vmapetr I see you were a contributor on the azure-pipelines-tasks-coverage-tools repo. Any chance this could help or be used at all?
